### ISSUE TITLE
sbt "++${{ scala-version }} scalafixAll --check" not parsed correctly

### ISSUE THREAD

**Issue by ruippeixotog** [scalacenter/scalafix#1488]
I'm trying to configure scalafix to run on CI on [PureConfig](https://github.com/pureconfig/pureconfig). We run one CI job per Scala version and the command we need to run is `sbt "++${{ matrix.scala-version }} scalafixAll --check"`. However, the `InputTask` that implements `scalafixAll` doesn't seem to be dealing with the `++` operator correctly:

```
[info] set current project to pureconfig (in build file:/Users/rui/Documents/Dev/own/pureconfig/)
[error] (core / scalafixAll) scalafix.sbt.InvalidArgument: 27 errors
[error] [E0] Unknown rule 'testkit/scalafixAll'
[error] [E1] Unknown rule 'tests/scalafixAll'
[error] [E2] Unknown rule 'bundle/scalafixAll'
[error] [E3] Unknown rule 'docs/scalafixAll'
[error] [E4] Unknown rule 'akka/scalafixAll'
[error] [E5] Unknown rule 'akka-http/scalafixAll'
[error] [E6] Unknown rule 'cats/scalafixAll'
(...)
```

This error doesn't happen if I run it without the version selector (`sbt "scalafixAll --check"`).

**Comment by bjaglin** [comment]
I am afraid this is an sbt limitation.

[There is a stale issue mentioning a similar issue for the `all` command upstream](https://github.com/sbt/sbt/issues/2439), so it seems that input tasks don't play well for at least 2 commands (`++` & `all`). I have tried to [tweak the syntax](https://github.com/bjaglin/pureconfig/pull/1), but ran into a wall.

The good news is that `+` does not seem to be impacted by this limitation, so instead of upstreaming the  `scalafixCheckAll` from https://github.com/pureconfig/pureconfig/pull/1188, I would rather recommend other users impacted by this (and unable to switch to the superior [sbt-projectmatrix](https://www.scala-sbt.org/1.x/docs/Cross-Build.html#Cross+building+a+project+using+sbt-projectmatrix) for cross-building) to use `+scalafixAll --check`. It comes with a few downsides as it won't allow a per-scala version view and requires full compilation of all sources (potentially in a separate, cache-cold CI workflow), but it works out of the box on all sbt/scalafix versions.

**Comment by mr-git** [comment]
Why is it wrong to have `scalafixCheckAll`, like we have `scalafmtCheckAll`? If SBT team didn't fix it in 8 years, why not provide alias in `scalafix`?

This SBT issue limits us to have command alias like: 
```sbt
addCommandAlias("check", "all scalafixCheckAll scalafmtCheckAll scalafmtSbtCheck")
```

The alternative I found so far:
```sbt
addCommandAlias("check", "scalafixAll --check; all scalafmtCheckAll scalafmtSbtCheck")
```
Though it will not help with CI build...

**Comment by bjaglin** [comment]
> Why is it wrong to have scalafixCheckAll, like we have scalafmtCheckAll?

A task alias can easily be defined on top of the input task:
```scala
override def projectSettings: Seq[Def.Setting[_]] =
    Seq(
      scalafixCheckAll := scalafixAll.toTask(" --check").value
    )
```

It can be done outside this repo (as a auto-plugin depending on ScalafixPlugin), but happy to take a PR that would add it here.

However, sbt-scalafix is more complex that sbt-scalafmt as it depends on compilation, so we'll need to decide how https://github.com/scalacenter/sbt-scalafix/pull/196 should behave when `scalafixCheckAll` is called, since exposing scalafix as a task opens the door for concurrent execution (via `all`) with a regular compile.

**Comment by bjaglin** [comment]
Also, even though that I would mean a difference with sbt-scalafmt, I would favor `scalafixAllCheck` since I see little value in adding `scalafixCheck`, but happy to hear other thoughts.

**Comment by mr-git** [comment]
Right, granting `scalafix` exclusive access to code is mandatory - I didn't think about this aspect.

Thank you!
